File: netty/ssy/netty_lecture/pythrift/PersonServiceImpl.py
# -*- coding:utf-8 -*-
import sys
reload(sys)
sys.setdefaultencoding('utf-8')

from py.thrift.generated import  ttypes

class PersonServiceImpl:

    def getPersonByUsername(self, username):
        print "Got client param: " + username

        person = ttypes.Person()
        person.username = username
        person.age = 20
        person.married = False
        return person

    def savePerson(self, person):
        print 'Got client param: '
        print person.username
        print person.age
        print person.married

File: netty/ssy/netty_lecture/src/main/java/cn/hunkier/thrift/PersonServiceImpl.java
package cn.hunkier.thrift;

import lombok.extern.slf4j.Slf4j;
import org.apache.thrift.TException;
import thrift.generated.DataException;
import thrift.generated.Person;
import thrift.generated.PersonService;

@Slf4j
public class PersonServiceImpl implements PersonService.Iface {
    @Override
    public Person getPersonByUsername(String username) throws DataException, TException {
        log.info("Got Client Param: " + username);
        Person person = new Person();
        person.setUsername(username);
        person.setAge(20);
        person.setMarried(false);
        return person;
    }

    @Override
    public void savePerson(Person person) throws DataException, TException {
        log.info("Got Client Param: ");
        log.info(person.getUsername());
        log.info(person.getAge()+"");
        log.info(person.isMarried()+"");
    }
}
